### DISCUSSION TITLE
Can't to use Naive on demand on Nuxt 3 Modules

### DISCUSSION THREAD

**Discussion by calebeaires** [tusen-ai/naive-ui#3720] [Q&A]
I have tried to use naive import on demand while creating a Nuxt 3 Module but I had no success. My aproach were a plugin system but when importing it does not work. 

```js
import {defineNuxtPlugin} from "#app";

import { NButton } from 'naive-ui';

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.vueApp.component(NButton.name, NButton)
})

```

```html
<template>
   <div>
       <n-button>This button does not work</n-button>
   </div>
</template>

```

I tried to setup Naive on this oficial [Nuxt 3 Module Starter](https://github.com/nuxt/starter/tree/module). The intention is not to use Naive on the Nuxt application itself but build an Nuxt Module that has some Naive components on its view.

The same use of case in with Element-Plus works pretty well.

**Comment by productdevbook** [comment]
Nuxt 3 Naive Module: https://github.com/huntersofbook/huntersofbook/tree/main/packages/naive-ui-nuxt

**Comment by jahnli** [comment]
We provide official nuxt plugin now. Check https://github.com/07akioni/naive-ui-nuxt-demo again.
